pkg/gds/emails/mock.go

```diff
@@ -241,12 +242,12 @@ func CheckEmails(t *testing.T, messages []*EmailMeta) {
 		sentEmails = append(sentEmails, msg)
 	}
 
-	for _, msg := range messages {
```

Comment:
@pdeziel These changes were necessary because the current code assumed there would only be one email sent to each contact per test, which isn't the case with the `TestCertManagerReissuance` test. I added a TODO to refactor this code in the future.

Reply:
I see, should we change that to a greater or equal check instead?

Reply:
I changed it to a GTE check but I let the TODO there, I think we could find a more elegant to handle this in the future.